tracking/src/DataCollection.js

```diff
@@ -1,24 +1,31 @@
 const NOT_AVAILABLE = "Unknown"
 class DataCollection {
     constructor() {
+    }
 
+    toJson() {
+        // update data first
         this.getLanguage(); 
         this.getScreenResolution(); 
-        this.getUrls(); 
-    }
+        this.getUrls();         
 
-    toJson() {
         const res = {}; 
         res.lang = this.language; 
         res.sr = this.resolution; 
-        res.url = this.url; 
-        res.r = this.referrer; 
+        res.url = this.url;
+        // override/remove user input for the following fields.
+        res.db = null; 
+        res.dm = null; 
+        res.os = null; 
+        res.osv = null; 
+        res.p = null; 
+        res.web_host = null; 
+        res.r = null;         
```

Comment:
referrer needs to be auto captured

Reply:
changed already. in the case that user input url directly in the url. there is no referrer. that will be null.